arkouda/array_api/_array_object.py

```diff
@@ -533,3 +774,11 @@ def size(self) -> int:
     @property
     def T(self) -> Array:
         raise ValueError("Not implemented")
+
+
+def implements_numpy(numpy_function):
+    """Register an __array_function__ implementation for MyArray objects."""
+    def decorator(func):
+        HANDLED_FUNCTIONS[numpy_function] = func
+        return func
+    return decorator
```

Comment:
In order for this to work, the Arkouda function header has to exactly match the NumPy function header, is that correct?

Reply:
That's right

Reply:
(one cavait: I think the header can be missing some optional arguments and it will still work as long as the caller doesn't use them)